075-connector-offsets-support.md

```diff
@@ -0,0 +1,270 @@
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-source-connector-offsets
+data:
+  offsets.json: |
+    {
+      "offsets": [
+        {
+          "partition": {
+            // Connector-defined source partition
+          },
+          "offset": {
+             // Connector-defined source offset
+          }
+        }
+      ]
+    }
+```
+
+For a sink connector it might look like:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-sink-connector-offsets
+data:
+  offsets.json: |
+    {
+      "offsets": [
+        {
+          "partition": {
+            "kafka_topic": "my-topic"
+              "kafka_partition": 2
+          },
+          "offset": {
+            "kafka_offset": 4
+          }
+        }
+      ]
+    }
+```
+
+Notes:
+* If the `listOffsets` property is missing from the KafkaConnector CR when the `strimzi.io/connectorOffsets=list` annotation is added the operator will update the KafkaConnector CR status to include a warning message:
+  ```yaml
+  - lastTransitionTime: "2024-06-04T08:44:15.913138115Z"
+    message: Failed to list the connector offsets due to missing property listOffsets in KafkaConnector CR.
+    reason: ListOffsets
+    status: "True"
+    type: Warning
+  ```
+  The operator will leave the annotation on the KafkaConnector resource until either the list operation succeeds, or the user removes the annotation.
+  This means the operator will retry the list on every reconciliation, allowing the condition to remain present for the user to see.
+
+### Altering offsets
+
+To alter offsets the user will annotate the KafkaConnector resource with the new annotation `strimzi.io/connectorOffsets` set to `alter`.
+After the annotation is added, on the next reconciliation the operator will call the PATCH /connectors/{connector}/offsets endpoint to alter the offsets.
+Once the patch is complete the operator will then remove the annotation from the KafkaConnector CR.
+
+The operator will read in the entry called `offsets.json` from the ConfigMap.
+The entry name matches the entry name used in the list operation, so that a user can list offsets, then edit the ConfigMap the operator created, then request the offsets be altered using that same ConfigMap.
+
+The name of the ConfigMap the operator will use to construct the request body will be set by the user in the KafkaConnector CR.
+The KafkaConnector CRD will be updated to contain a new property called `alterOffsets` that must be set for the `strimzi.io/connectorOffsets=alter` annotation to take effect.
+The structure of the `alterOffsets` property will be as below, to allow it to be extended in future if required:
+
+```yaml
+alterOffsets:
+  fromConfigMap:
+    name: my-connector-offsets
+```
+
+Notes:
+* The format of the request body will be validated to make sure it is valid JSON, but the operator won't perform any further validation, instead relying on the API endpoint to return a reasonable error message.
```

Comment:
1. What does "valid JSON" mean here? I guess it must mean "syntactically valid JSON", because source connectors have an unknown schema for their offset JSON. But it would be good to be explicit in the doc. 
2. Let's also be explicit that `offsets.json` being absent from the ConfigMap is also treated as an error. 
3. I suppose there's also no harm in saying that it's not an error if the ConfigMap contains keys others than `offsets.json`

Reply:
1. Correct, I've added the word "syntactically" to be more explicit.
2. Added
3. Added